src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -277,10 +277,12 @@ private IEnumerable<DiagnosticAnalyzer> GetAnalyzersForTypeNames(Assembly analyz
                 Type type;
                 try
                 {
-                    type = Type.GetType(typeName + ", " + analyzerAssembly.FullName, throwOnError: true);
+                    type = analyzerAssembly.GetType(typeName) ?? 
```

Comment:
Consider documenting why we do things this way.  i.e. "analyzer.GetType" is the most robust method (since it doesn't need binding paths and all that goop).  But if it fails, you get nothing but "null".  Then the rest of the code can try to get a decent error if possible.